syntax = "proto3";

package litepb.test.interop;

message BasicScalars {
    int32 int32_field = 1;
    string string_field = 2;
}

message AllScalars {
  int32 v_int32 = 1;
  int64 v_int64 = 2;
  uint32 v_uint32 = 3;
  uint64 v_uint64 = 4;
  bool v_bool = 5;
  string v_string = 6;
  bytes v_bytes = 7;
  float v_float = 8;
  double v_double = 9;
}

message SignedInts {
  sint32 v_sint32 = 1;
  sint64 v_sint64 = 2;
}

message FixedTypes {
  fixed32 v_fixed32 = 1;
  fixed64 v_fixed64 = 2;
  sfixed32 v_sfixed32 = 3;
  sfixed64 v_sfixed64 = 4;
}

message RepeatedFields {
  repeated int32 packed_ints = 1 [packed=true];
  repeated string strings = 2;
}

message MapField {
  map<string, int32> items = 1;
}

message Outer {
  message Inner {
    int32 value = 1;
  }
  Inner inner = 1;
  string name = 2;
}

message OneofTest {
  oneof choice {
    int32 int_value = 1;
    string string_value = 2;
  }
}

enum TestEnum {
  UNKNOWN = 0;
  FIRST = 1;
  SECOND = 2;
}

message EnumTest {
  TestEnum value = 1;
}
